numba/np/npdatetime.py

```diff
@@ -810,6 +810,10 @@ def _cast_npdatetime_int64(context, builder, fromty, toty, val):
 @overload_method(types.NPTimedelta, '__hash__')
 @overload_method(types.NPDatetime, '__hash__')
 def ol_hash_npdatetime(x):
+    if numpy_support.numpy_version >= (2, 2)\
+          and x.name == 'timedelta64[]':
```

Comment:
Is matching on the `dtype` perhaps more reliable than the string representation?